src/libstd/path.rs

```diff
@@ -1398,13 +1398,17 @@ impl<'a> From<&'a Path> for Box<Path> {
 
 #[stable(feature = "path_buf_from_box", since = "1.18.0")]
 impl From<Box<Path>> for PathBuf {
+    /// Converts a `Box<Path>` into a `PathBuf`.
+    /// This conversion does not allocate memory
     fn from(boxed: Box<Path>) -> PathBuf {
         boxed.into_path_buf()
     }
 }
 
 #[stable(feature = "box_from_path_buf", since = "1.20.0")]
 impl From<PathBuf> for Box<Path> {
+    /// Converts a `PathBuf` into a `Box<Path>`.
+    /// This conversion does not allocate memory
```

Comment:
Question for @rust-lang/docs: should we start specifying allocation behavior in the docs, like this line? I forgot if we've discussed this already in the past. Also FWIW, doesn't look like we specify allocation behavior in the [docs for the underlying `into_boxed_path` method](https://doc.rust-lang.org/nightly/std/path/struct.PathBuf.html#method.into_boxed_path).

Reply:
I start this would be a good idea. Rust is meant to be memory efficient, therefore, having allocation behavior precisions seems like a must have.

Reply:
I think in many cases we'll need to check in with @rust-lang/libs but in obvious cases like this, it's meant to do so and so should be.

Reply:
Convertion from an extensible container like `PathBuf` to a fixed size like `Box<Path>` doesn't necessarily explicitly allocate memory but it does use `shrink_to_fit` which attempts to shrink the allocation of `PathBuf` to precisely match the `Path` in question. Most of the time this doesn't allocate memory, but to be safe it's probably best to avoid accidentally providing a guarantee that this doesn't allocate.

Reply:
@alexcrichton  Should we add this in the docs or remove them all together?

Reply:
Nah I think having them is fine so long as it's clear that *currently* it's unlikely to reallocate/allocate memory, but this isn't necessarily guaranteed in the future